lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py

```diff
@@ -0,0 +1,708 @@
+#!/usr/bin/python
+
+# (c) 2018, NetApp, Inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: netapp_e_mgmt_interface
+short_description: NetApp E-Series management interface configuration
+description:
+    - Configure the E-Series management interfaces
+version_added: '2.7'
+author:
+    - Michael Price (@lmprice)
+    - Nathan Swartz (@ndswartz)
+extends_documentation_fragment:
+    - netapp.eseries
+options:
+    state:
+        description:
+            - Enable or disable IPv4 network interface configuration.
+            - Either IPv4 or IPv6 must be enabled otherwise error will occur.
+            - Only required when enabling or disabling IPv4 network interface
+        choices:
+            - enable
+            - disable
+        required: no
+        aliases:
+            - enable_interface
+    controller:
+        description:
+            - The controller that owns the port you want to configure.
+            - Controller names are represented alphabetically, with the first controller as A,
+             the second as B, and so on.
+            - Current hardware models have either 1 or 2 available controllers, but that is not a guaranteed hard
+             limitation and could change in the future.
+        required: yes
+        choices:
+            - A
+            - B
+    name:
+        description:
+            - The port to modify the configuration for.
+            - The list of choices is not necessarily comprehensive. It depends on the number of ports
+              that are present in the system.
+            - The name represents the port number (typically from left to right on the controller),
+              beginning with a value of 1.
+            - Mutually exclusive with I(channel).
```

Comment:
You can add the following to `argument_spec`
`mutually_exclusive=(['name', 'channel'],),`